python/python/bystro/prs/preprocess_for_prs.py

```diff
@@ -209,64 +267,78 @@ def calculate_abs_effect_weights(merged_scores_genos: pd.DataFrame) -> pd.DataFr
     ]
 
 
-def clean_dosage_for_analysis(merged_scores_genos: pd.DataFrame, column_to_drop: str) -> pd.DataFrame:
-    """Drop missing values and adjust scores."""
-    column_to_drop_index = merged_scores_genos.columns.get_loc(column_to_drop)
-    columns_to_keep = [
-        "allele_comparison",
-        *list(merged_scores_genos.columns[column_to_drop_index + 1 :]),
-    ]
-    genos_wo_extra_cols = merged_scores_genos[columns_to_keep]
-    genos_wo_missing = genos_wo_extra_cols.dropna()
-    allele_comparison_results = genos_wo_missing.apply(
-        compare_alleles, col1="index", col2="ID_effect_as_alt", axis=1
+def clean_scores_for_analysis(
+    max_effect_per_bin: pd.DataFrame, column_to_drop: str
+) -> tuple[pd.DataFrame, pd.DataFrame]:
+    """Drop extra columns and prepare df with final set of loci for dosage matrix filtering."""
+    scores_overlap_adjusted = max_effect_per_bin.drop(columns=["bin", "abs_effect_weight"])
+    scores_overlap_adjusted = scores_overlap_adjusted.set_index("SNPID")
+    format_col_index = scores_overlap_adjusted.columns.get_loc(column_to_drop)
+    columns_to_keep = ["allele_comparison"] + list(
+        scores_overlap_adjusted.columns[format_col_index + 1 :]
     )
-    genos_adjusted = allele_comparison_results.apply(adjust_dosages, axis=1)
-    genos_wo_nomatch_alleles = genos_adjusted[
-        genos_adjusted["allele_comparison"] != "alleles do not match"
-    ]
-    return genos_wo_nomatch_alleles.iloc[:, 1:]
+    loci_and_allele_comparison = scores_overlap_adjusted[columns_to_keep]
+    return scores_overlap_adjusted, loci_and_allele_comparison
 
 
-def ld_clump(merged_scores_genos: pd.DataFrame, map_directory_path: str) -> pd.DataFrame:
-    """Bin using genetic map, clump, and adjust dosages."""
+def ld_clump(scores_overlap: pd.DataFrame, map_directory_path: str) -> tuple[pd.DataFrame, pd.DataFrame]:
+    """Bin using genetic map, clump, and adjust dosages and determine if effect allele is alt or ref."""
     bin_mappings = _preprocess_genetic_maps(map_directory_path)
-    merged_scores_genos_w_bins = assign_bins(merged_scores_genos, bin_mappings)
-    merged_scores_genos_abs_val = calculate_abs_effect_weights(merged_scores_genos_w_bins)
-    max_effect_per_bin = select_max_effect_per_bin(merged_scores_genos_abs_val)
-    return clean_dosage_for_analysis(max_effect_per_bin, "ID_effect_as_ref")
+    scores_overlap = scores_overlap.reset_index()
+    allele_comparison_results = scores_overlap.apply(
+        compare_alleles, col1="SNPID", col2="ID_effect_as_alt", axis=1
+    )
+    scores_overlap.insert(0, "allele_comparison", allele_comparison_results)
+    scores_overlap_w_bins = assign_bins(scores_overlap, bin_mappings)
+    scores_overlap_abs_val = calculate_abs_effect_weights(scores_overlap_w_bins)
+    max_effect_per_bin = select_max_effect_per_bin(scores_overlap_abs_val)
+    return clean_scores_for_analysis(max_effect_per_bin, "ID_effect_as_ref")
+
+
+def extract_clumped_thresholded_genos(
```

Comment:
this will not work as written for large dataframes because even for the subset of loci, we will be unable to always load this into memory; we'll want to perform PRS on groups of samples, and save out just the PRS scores per sample